libs/langgraph/langgraph/utils/cache.py

```diff
@@ -0,0 +1,26 @@
+from collections.abc import Hashable
+from typing import Any
+
+
+def _freeze(obj: Any) -> Hashable:
+    if isinstance(obj, dict):
+        # sort keys so {"a":1,"b":2} == {"b":2,"a":1}
+        return tuple(sorted((k, _freeze(v)) for k, v in obj.items()))
+    elif isinstance(obj, (list, tuple, set, frozenset)):
+        return tuple(_freeze(x) for x in obj)
+    # numpy / pandas etc. can provide their own .tobytes()
+    elif hasattr(obj, "tobytes"):
+        return (
+            type(obj).__name__,
+            obj.tobytes(),
+            obj.shape if hasattr(obj, "shape") else None,
+        )
+    return obj  # strings, ints, dataclasses with frozen=True, etc.
+
+
+def default_cache_key(*args: Any, **kwargs: Any) -> bytes:
```

Comment:
```suggestion
def default_cache_key_generator(*args: Any, **kwargs: Any) -> bytes:
```

Or something to indicate it's a callable